### PR TITLE
Update dbt_project.yml.md

### PR DESCRIPTION
**Pull request by mirnawong1**
this pr adds additiona clarification on the dash and underscores rules for the `dbt_project.yml` and other configs. [relevant slack thread](https://dbt-labs.slack.com/archives/C05FWBP9X1U/p1716268955554979?thread_ts=1715189163.754789&cid=C05FWBP9X1U)

### CODE REVIEW

File: website/docs/reference/dbt_project.yml.md [hunk 1]
```diff
@@ -242,3 +244,32 @@ vars:
...
+## Naming convention
+
+It's important to follow the correct naming conventions for the YAML configs in your `dbt_project.yml` file to make sure dbt can understand them properly. This is especially true for resource types with more than one word:
+
+- Use dashes (`-`) when configuring resource types with multiple words in your `dbt_project.yml` file:
+
+  <File name="dbt_project.yml" >
+  
+    ```yaml
+    [saved-queries](/docs/build/saved-queries#project-level-saved-queries): ## This resource type in the dbt_project.yml uses dashes
```
Comment (mirnawong1) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
    [saved-queries](/docs/build/saved-queries#project-level-saved-queries): ## Uses dashes in the dbt_project.yml file.
```

File: website/docs/reference/dbt_project.yml.md [hunk 2]
```diff
@@ -242,3 +244,32 @@ vars:
...
+## Naming convention
+
+It's important to follow the correct naming conventions for the YAML configs in your `dbt_project.yml` file to make sure dbt can understand them properly. This is especially true for resource types with more than one word:
+
+- Use dashes (`-`) when configuring resource types with multiple words in your `dbt_project.yml` file:
+
+  <File name="dbt_project.yml" >
+  
+    ```yaml
+    [saved-queries](/docs/build/saved-queries#project-level-saved-queries): ## Uses dashes in the dbt_project.yml file.
```
Comment (mirnawong1) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
    [saved-queries](/docs/build/saved-queries#project-level-saved-queries): ## Use dashes in the dbt_project.yml file.
```

File: website/docs/reference/dbt_project.yml.md [hunk 3]
```diff
@@ -242,3 +244,32 @@ vars:
...
+            enabled: true
+    ```
+  </File>
+
+- Use underscore (`_`) when configuring resource types with multiple words outside the `dbt_project.yml` file:
+
+  <File name="models/semantic_models.yml" >
+  
+    ```yaml
+    [saved_queries](/docs/build/saved-queries#configure-saved-query): ## This example uses underscores
```
Comment (mirnawong1) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
    [saved_queries](/docs/build/saved-queries#configure-saved-query): ## Use underscores everywhere outside the dbt_project.yml file.
```

File: website/docs/docs/build/saved-queries.md [hunk 4]
```diff
@@ -119,6 +119,8 @@ saved-queries:
 ```
 </File>
 
+For more info on `dbt_project.yml` and config naming conventions, check out [naming convention](/reference/dbt_project.yml).
```
Comment (matthewshaver) [inline website/docs/docs/build/saved-queries.md]: "Naming convention" isn't mentioned anywhere on the ref page, so updating the description for accuracy.
```suggestion
For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml).
```

File: website/docs/docs/build/semantic-models.md [hunk 5]
```diff
@@ -144,6 +144,8 @@ Semantic models support [`meta`](/reference/resource-configs/meta), [`group`](/r
         some_key: some_value
   ```
 
+For more info on `dbt_project.yml` and config naming conventions, check out [naming convention](/reference/dbt_project.yml).
```
Comment (matthewshaver) [inline website/docs/docs/build/semantic-models.md]:
```suggestion
For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml).
```

File: website/docs/reference/dbt_project.yml.md [hunk 6]
```diff
@@ -1,23 +1,25 @@
...
 
 </VersionBlock>
 
 <VersionBlock firstVersion="1.5">
 
-By default, dbt will look for `dbt_project.yml` in your current working directory and its parents, but you can set a different directory using the `--project-dir` flag or the `DBT_PROJECT_DIR` environment variable.
+- By default, dbt looks for the `dbt_project.yml` in your current working directory and its parents, but you can set a different directory using the `--project-dir` flag or the `DBT_PROJECT_DIR` environment variable.
 
-Starting from dbt v1.5 and higher, you can specify your dbt Cloud project ID in the `dbt_project.yml` file using `project-id` under the `dbt-cloud` config. To find your project ID, check your dbt Cloud project URL, such as `https://cloud.getdbt.com/11/projects/123456`, where the project ID is `123456`.
+- Specify your dbt Cloud project ID in the `dbt_project.yml` file using `project-id` under the `dbt-cloud` config. Find your project ID in your dbt Cloud project URL: In example `https://cloud.getdbt.com/11/projects/123456`, the project ID is `123456`.
```
Comment (matthewshaver) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
- Specify your dbt Cloud project ID in the `dbt_project.yml` file using `project-id` under the `dbt-cloud` config. Find your project ID in your dbt Cloud project URL: For example, in `https://cloud.getdbt.com/11/projects/123456`, the project ID is `123456`.
```

File: website/docs/reference/dbt_project.yml.md [hunk 7]
```diff
@@ -242,3 +244,34 @@ vars:
 </File>
 
 </VersionBlock>
+
+## Naming convention
+
+It's important to follow the correct naming conventions for the YAML configs in your `dbt_project.yml` file to make sure dbt can understand them properly. This is especially true for resource types with more than one word.
```
Comment (matthewshaver) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
It's important to follow the correct YAML naming conventions for the configs in your `dbt_project.yml` file to ensure dbt can process them properly. This is especially true for resource types with more than one word.
```

File: website/docs/reference/dbt_project.yml.md [hunk 8]
```diff
@@ -242,3 +244,34 @@ vars:
...
+## Naming convention
+
+It's important to follow the correct YAML naming conventions for the configs in your `dbt_project.yml` file to ensure dbt can process them properly. This is especially true for resource types with more than one word.
+
+- Use dashes (`-`) when configuring resource types with multiple words in your `dbt_project.yml` file. Here's an example for [saved queries](/docs/build/saved-queries#configure-saved-query):
+
+    <File name="dbt_project.yml">
+
+    ```yml
+    saved-queries:  # Use dashes in the dbt_project.yml file.
```
Comment (matthewshaver) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
    saved-queries:  # Use dashes for resource types in the dbt_project.yml file.
```

File: website/docs/docs/build/saved-queries.md [hunk 9]
```diff
@@ -119,6 +119,8 @@ saved-queries:
 ```
 </File>
 
+For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml).
```
Comment (matthewshaver) [inline website/docs/docs/build/saved-queries.md]:
```suggestion
For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml#naming-convention).
```

File: website/docs/docs/build/semantic-models.md [hunk 10]
```diff
@@ -144,6 +144,8 @@ Semantic models support [`meta`](/reference/resource-configs/meta), [`group`](/r
         some_key: some_value
   ```
 
+For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml).
```
Comment (matthewshaver) [inline website/docs/docs/build/semantic-models.md]:
```suggestion
For more information on `dbt_project.yml` and config naming conventions, see the [dbt_project.yml reference page](/reference/dbt_project.yml#naming-convention).
```

File: website/docs/reference/dbt_project.yml.md [hunk 11]
```diff
@@ -242,3 +244,34 @@ vars:
...
+    ```yml
+    saved-queries:  # Use dashes for resource types in the dbt_project.yml file.
+      my_saved_query:
+        config:
+          +cache:
+            enabled: true
+    ```
+    </File>
+
+- Use underscore (`_`) when configuring resource types with multiple words outside the `dbt_project.yml` file:
```
Comment (matthewshaver) [inline website/docs/reference/dbt_project.yml.md]:
```suggestion
- Use underscore (`_`) when configuring resource types with multiple words for YAML files other than the `dbt_project.yml` file. For example, here's the same saved queries resource in the `semantic_models.yml` file:
```

**Review by matthewshaver — APPROVED** [review]

**Pull request merged**